addons/account/models/account_journal.py

```diff
@@ -798,3 +781,17 @@ def _get_last_bank_statement(self, domain=None):
         last_statement_domain = (domain or []) + [('journal_id', '=', self.id)]
         last_st_line = self.env['account.bank.statement.line'].search(last_statement_domain, order='date desc, id desc', limit=1)
         return last_st_line.statement_id
+
+    def _ensure_payment_method_type_on_journal(self, payment_method_type):
+        for journal in self:
+            methods = journal.outbound_payment_method_ids if payment_method_type.payment_type == "outbound" else journal.inbound_payment_method_ids
+            if not any(method.code == payment_method_type.code for method in methods):
+                method = self.env['account.payment.method'].create({
+                    'method_type': payment_method_type.id,
+                    'journal_id': journal.id
+                })
+
+                if payment_method_type.payment_type == "outbound":
+                    journal.outbound_payment_method_ids += method
+                else:
+                    journal.inbound_payment_method_ids += method
```

Comment:
You don't need to assign specificaly your record to a one2many. This is made automatically by the orm.